### PR TITLE
[H] Improved positioning of leading and trailing widgets in overflowing ListTiles

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Hixie** [comment]
cc @HansMuller

**Comment by Hixie** [comment]
There's still an issue with using ListTile with overflowing title/subtitle widgets in that the results end up looking kinda weird because of this code:
https://github.com/flutter/flutter/pull/17496/files#diff-53f33273ae4e7462729c5f4b7394428bR778

**Comment by andyduke** [comment]
There is not enough ability to control the alignment of the leading and trailing, sometimes it is necessary to set the alignment of the leading to the top, for ListTile without a subtitle, but with a multi-line title.

Maybe add leadingAlignment and trailingAlignment properties with auto, top, center, bottom values?

**Comment by Hixie** [comment]
@andyduke Just make a custom widget at that point. ListTile is really just a Row that happens to match the material spec's redlines.

**Comment by Hixie** [comment]
cc @HansMuller for review.

**Review by jacob314 — APPROVED** [review]
lgtm

File: packages/flutter/lib/src/material/list_tile.dart [hunk 1]
```diff
@@ -246,27 +245,45 @@ class ListTile extends StatelessWidget {
...
   /// Typically a [Text] widget.
+  ///
+  /// This should not wrap.
   final Widget title;
 
   /// Additional content displayed below the title.
   ///
   /// Typically a [Text] widget.
+  ///
+  /// If [isThreeLine] is false, this should not wrap.
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/material/list_tile.dart]: I went to do this, but I think actually that might make it less clear. They're cases, they're instructions for the person providing the `subtitle`.

File: packages/flutter/lib/src/material/list_tile.dart [hunk 2]
```diff
@@ -966,8 +985,34 @@ class _RenderListTile extends RenderBox {
...
+    double trailingY;
+    if (isOneLine) {
+      leadingY = (defaultTileHeight - leadingSize.height) / 2.0;
+      trailingY = (defaultTileHeight - trailingSize.height) / 2.0;
+    } else if (isTwoLine) {
+      if (isDense) {
+        leadingY = 12.0; // by extrapolation
+        trailingY = 20.0; // by extrapolation
+      } else {
+        if (leadingSize.height <= 40.0) {
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/material/list_tile.dart]: agreed, done

**Comment by Hixie** [comment]
Thanks for the review. Updated. Will land on green.

**Pull request merged**